sound/soc/sof/intel/hda.c

```diff
@@ -1348,7 +1366,7 @@ struct snd_soc_acpi_mach *hda_machine_select(struct snd_sof_dev *sdev)
 			tplg_filename = devm_kasprintf(sdev->dev, GFP_KERNEL,
 						       "%s%s%d%s",
 						       sof_pdata->tplg_filename,
-						       "-dmic",
+						       i2s_mach_found ? "-dmic" : "-",
```

Comment:
that's almost the same code as line 580, so there's likely additional simplifications possible.